src/Apple/Provider.php

```diff
@@ -101,12 +116,53 @@ public function getAccessTokenResponse($code)
      */
     protected function getUserByToken($token)
     {
-        static::verify($token);
+        $this->checkToken($token);
         $claims = explode('.', $token)[1];
 
         return json_decode(base64_decode($claims), true);
     }
 
+    protected function getClientSecret()
+    {
+        if (!$this->jwtConfig) {
+            $this->getJwtConfig(); // Generate Client Secret from private key if not set.
+        }
+
+        return $this->clientSecret;
+    }
+
+    protected function getJwtConfig()
+    {
+        if (!$this->jwtConfig) {
+            $private_key_path = config('services.apple.private_key', '');
+            $private_key_passphrase = config('services.apple.passphrase', '');
+            $signer = config('services.apple.signer', '');
+
+            if (empty($signer) || !class_exists($signer)) {
+                $signer = !empty($private_key_path) ? \Lcobucci\JWT\Signer\Ecdsa\Sha256::class : AppleSignerNone::class;
+            }
+            
+            $this->privateKey = $private_key_path; // Support for plain text private keys
+
+            if (!empty($private_key_path) && file_exists($private_key_path)) {
+                $this->privateKey = file_get_contents($private_key_path);
+            }
+
+            $this->jwtConfig = Configuration::forSymmetricSigner(
```

Comment:
Just to confirm, there is no bc break here, existing integrations using the secret key will continiue to work fine?

Reply:
yes, if you pass the path to the private key file, it will get the contents of the private key file, otherwise you put the private key contents by default.